CMakeLists.txt

```diff
@@ -133,6 +133,7 @@ if (TI_WITH_BACKTRACE)
   add_subdirectory(external/backward_cpp)
 endif()
 
+include(cmake/Distributed.cmake)
```

Comment:
In the complete state, will this `Distributed Compilation` only apply to `Taichi CI System` and individual contributors won't be able to use it? 

If it's not designed for individual contributors, then we probably want to add a check and give warnings if it's not running on CI (just in case people mistakenly set DISTRIBUTED_COMPILE=ON)